lock implementation for runtime resource publishing

fadedreamz opened the pull request:
Signed-off-by: fadedreamz <fadedreamz@gmail.com>

fadedreamz committed:
lock implementation for runtime resource publishing

Signed-off-by: fadedreamz <fadedreamz@gmail.com>

fadedreamz commented:
@ben-crowhurst please have a look and give your suggestion

ben-crowhurst commented:
Tests are failing.

fadedreamz committed:
updated acceptance test to reflect runtime service publishing

Signed-off-by: fadedreamz <fadedreamz@gmail.com>

ben-crowhurst commented:
Thanks looks good.  I'll pull it into master once we've tag the 4.5 release as its currently feature frozen.

fadedreamz commented:
@ben-crowhurst glad to hear that

Commit:
HTTP Pipelining, resolved #23.

ben-crowhurst committed:
Build script clean-up.

ben-crowhurst committed:
Resolved template deduction issues.

Commit:
Resolved Windows build warnings.

ben-crowhurst committed:
Silenced OSX build warning.

Commit:
Silenced VS C++ warnings.

Commit:
Design Documentation.

Commit:
Resolved DESIGN.md layout issues.

Commit:
Resolved DESIGN.md layout issues.

Commit:
Reworked documentation.

Commit:
Update to Design documentation.

Commit:
Fleshing out API documentation.

Commit:
Service API documentation.

Commit:
Session API documentation.

Commit:
SessionManager API documentation.

Commit:
Uri API documentation.

Commit:
Settings API documentation.

Commit:
Completed version 4.5 API documentation.

Commit:
Update readme.

nunojpg committed:
Remove defunct sessions

When the browser closes the connection or there is a network drop
the session is not removed which is would cause defunct sessions
to not be reclaimed causing a memory leak.

Since there is not a specific callback for a session close event,
use the periodic callbcack to do this housekeeping.

Signed-off-by: Nuno Goncalves <nunojpg@gmail.com>

ben-crowhurst committed:
Updated copyright notice.

ben-crowhurst committed:
Resolved linking issue #176

ben-crowhurst committed:
Bumped copyright notice.

nunojpg committed:
Add Clang build

Signed-off-by: Nuno Goncalves <nunojpg@gmail.com>

nunojpg committed:
Fix HTTPS example link

eranhober commented:
Hi there, please note that the ServiceImpl::lock::write should be created under the suppress function as well so that at runtime both publish and suppress functionality will be available.
Regards,

Eran

Commit:
modified all the CMAKE_SOURCE_DIR to CMAKE_CURRENT_SOURCE_DIR

Commit:
Changed CMAKE_CURRENT_SOURCE_DIR to PROJECT_SOURCE_DIR based on Issue #186

ben-crowhurst committed:
MacOS CI support.

ben-crowhurst committed:
allow hyphen and underscore in path parameter names.

ben-crowhurst committed:
Renamed test case.

nunojpg committed:
Do not specify openssl path on Travis config

Path is already on cmake module.

Signed-off-by: Nuno Goncalves <nunojpg@gmail.com>

nunojpg committed:
Specify project language for cmake

Signed-off-by: Nuno Goncalves <nunojpg@gmail.com>

clrusby committed:
Correct the string for IPv4 local/remote endpoints

ben-crowhurst committed:
Regression test covering PR #207, incorrect IPv4 endpoint string.

ben-crowhurst committed:
Inclusion of AppVeyor badage.

ben-crowhurst committed:
Version control AppVeyor configuration.

Commit:
Added isOpen check in read and write functions for socket_impl.cpp

ben-crowhurst committed:
Ability to parse float/double from request data.

ben-crowhurst committed:
Updated with ready handler API.

aberaud committed:
Win32: use _WIN32 macro, proper case for winsock2.h

Fix various issues preventing compilation in some environments
(e.g. cross-compiling with mingw) on a case-sensitive system,
and use the official _WIN32 macro to detect the Win32 environment as WIN32 is not always defined.

aberaud committed:
Handle DNS resolution error

DNS resolution error currently causes
indefinite hangup as the callback is
not called and higher-level layers
are not informed.

This commit bubbles up the error,
allowing the request to fail.

ben-crowhurst committed:
Revert "Added isOpen check in read and write functions for socket_impl.cpp"

ben-crowhurst committed:
Resolved issue reported in PR227.

ben-crowhurst committed:
Coding standards violation resolved.

ben-crowhurst committed:
Coding standards violation resolved.

ben-crowhurst committed:
Resolved issue #172.

ben-crowhurst committed:
Resolved issue #171.

ben-crowhurst committed:
Improved bug resolution #171.

ben-crowhurst committed:
Updated dependency chain.

ben-crowhurst committed:
Bumped build version for 4.6 release.

ben-crowhurst committed:
Resolved issue #223.

aberaud committed:
http: check for locale before duplicating string

Commit:
Moved websocket write buffer to a locally scoped variable.

Commit:
Resolved issue #235.

Commit:
Silenced Window build warnings.

Commit:
Resolved broken test case on FreeBSD.

ben-crowhurst committed:
Resolved SEGFAULT with OpenSSL and ASIO strands.

Commit:
removed wrongly placed `using` from docs

It's invalid http://cpp.sh/6dawz

Commit:
fixed `#include <restbed>` error on `add_subdirectory()`

abogani committed:
Replace include guards with pragma once

mipac commented:
hi, any news of this feature ?
